test/GPv2AllowanceManager.test.ts

```diff
@@ -1,18 +1,20 @@
 import IERC20 from "@openzeppelin/contracts/build/contracts/IERC20.json";
 import { expect } from "chai";
-import { Contract, BigNumberish } from "ethers";
+import { Contract } from "ethers";
 import { artifacts, ethers, waffle } from "hardhat";
 
-interface Transfer {
-  owner: string;
-  token: string;
-  amount: BigNumberish;
-}
+import { ExecutedTrade, composeExecutedTrade } from "./GPv2TradeExecution.test";
 
 function composeTransfers(
-  transfers: Transfer[],
-): [string, string, BigNumberish][] {
-  return transfers.map(({ owner, token, amount }) => [owner, token, amount]);
+  trades: Pick<ExecutedTrade, "owner" | "sellToken" | "sellAmount">[],
```

Comment:
What about doing something like this instead:
```
interface ExecutedSell {
  owner: string;
  sellToken: string;
  sellAmount: BigNumber;
}
interface ExecutedBuy {
  owner: string;
  buyToken: string;
  buyAmount: BigNumber;
}
type ExecutedTrade = ExecutedBuy & ExecutedSell;
```

Reply:
Nice suggestion! What about just naming the `Executed{Sell, Buy}` types?:
```ts
type ExecutedSell = Pick<ExecutedTrade, "owner" | "sellToken" | "sellAmount">;
type ExecutedBuy = Pick<ExecutedTrade, "owner" | "buyToken" | "buyAmount">;
```

My main argument for this is that this keeps the `ExecutedTrade` struct mapped directly with the Solidity `GPv2TradeExecution.Data` type.

Reply:
Naming the structs and moving them close to `ExecutedTrade` works for me. It was a bit weird to see such specific typing operations happening here.

Reply:
I called it `{In, Out}Transfer` to match what we had before and the names of the functions in #227